templates/profile.php

```diff
@@ -14,14 +14,15 @@ function pmpromd_profile_preheader()
 		$main_post_id = $post->ID;
 
 		//Get the profile user
-		if(!empty($_REQUEST['pu']) && is_numeric($_REQUEST['pu']))
-			$pu = get_user_by('id', $_REQUEST['pu']);
-		elseif(!empty($_REQUEST['pu']))
-			$pu = get_user_by('slug', $_REQUEST['pu']);
-		elseif(!empty($current_user->ID))
+		$user_identifier = apply_filters( 'pmpromd_user_identifier', 'user_nicename' );
+
+		if( !empty( $_REQUEST['pu'] ) ) {
+			$pu = get_user_by( $user_identifier, $_REQUEST['pu'] );

```

Comment:
Let's abstract the `get_user_by` handling into a new function that looks it up based on the user identifier. It should take one parameter which is the value to look up by.

Ultimately, this line would look something like this (keep the other logic in the else block the same).

```suggestion
			$pu = pmpromd_get_user_by_identifier( sanitize_text_field( $_REQUEST['pu'] ) );
```

And the function itself might look like:

```php
function pmpromd_get_user_by_identifier( $value ) {
	$user_identifier = pmpromd_get_user_identifier();

	return get_user_by( $user_identifier, $value );
}
```